### PR TITLE
renamed getUserByEmail to getUserbyUsernameOrEmail and modified the t…

### PR DESCRIPTION
**Pull request by fanyui**
…ests that depend on it as well as the exception that it throws

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [x ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by fanyui** [comment]
@wluyima  This fixes the issue with triggering the reset in the websevice with a post with either email or username

File: api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -718,17 +718,20 @@ public String getSecretQuestion(User user) throws APIException {
...
 	@Transactional(readOnly = true)
-	public User getUserByEmail(String email) {
-		if (StringUtils.isNotBlank(email)) {
-			return dao.getUserByEmail(email);
+	public User getUserByUsernameOrEmail(String usernameOrEmail) {
+		if (StringUtils.isNotBlank(usernameOrEmail)) {
+			User user = dao.getUserByEmail(usernameOrEmail);
+			if (user == null) {
+				return getUserByUsername(usernameOrEmail);
+			} else {
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java]: This else clause is redundant since the else clause always returns, just return the user

File: api/src/main/resources/liquibase-update-to-2.2.xml [hunk 2]
```diff
@@ -269,11 +269,15 @@
 		<preConditions onFail="MARK_RAN">
 			<not>
 				<columnExists tableName="users" columnName="activation_key"/>
+				<columnExists tableName="users" columnName="email"/>
```
Comment (wluyima) [inline api/src/main/resources/liquibase-update-to-2.2.xml]: Each column must be added in a separate changeset
